packages/firestore/src/api/database.ts

```diff
@@ -552,6 +553,41 @@ export class WriteBatch
   }
 }
 
+class UntypedFirestoreDataConverterAdapter<U>
```

Comment:
This now looks like the rest of our "Compat" classes. You could follow the existing pattern and start this off as such:

```
class FirestoreDataConverter<U>
  extends Compat<ExpFirestoreDataConverter<U>>
  implements FirestoreDataConverter<T> {
```

I also wonder if thanks to this adapter, we may now be able to stop using the `UntypedFirestoreDataConverterAdapter`.

Reply:
Good point about using the `Compat` class.

I was also thinking the same thing about `UntypedFirestoreDataConverterAdapter` not being needed anymore. I think that you're right, if we use this "compat" adapter strategy then we can get rid of `UntypedFirestoreDataConverterAdapter`.